src/ol/webgl/TileTexture.js

```diff
@@ -80,20 +80,26 @@ function uploadDataTexture(
   }
 
   let format;
+  let internalFormat;
   switch (bandCount) {
     case 1: {
       format = gl.LUMINANCE;
+      textureType = gl.UNSIGNED_BYTE;
+      data = new Uint8Array(Array.prototype.slice.call(data));
+
       break;
     }
     case 2: {
       format = gl.LUMINANCE_ALPHA;
       break;
     }
     case 3: {
+      internalFormat = data instanceof Float32Array ? RGB32F : gl.RGB;
```

Comment:
Many PCs now running Windows 10 (upgraded from Windows 7) have no WebGL2 support but do support `GL_OES_texture_float` and `GL_OES_texture_float_linear`.

Reply:
Thank you, I have made a fix of it. Now it supports both webgl and webgl2.